Configure prettier for the project

HenrikeW opened the pull request:
## Related Issues
This PR closes issue #38 .

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## Changes
- add Prettier as a dependency
- add a configuration file for Prettier
- add a `prettierignore` file
- add documentation on how to use Prettier from the command line
- format all files in the repo

## Testing
- Make sure you can use Prettier from the command line as described in the readme
- check the `prettierignore` file: does it make sense?

## Further comments
The formatting configurations are based on a suggestion I found in Prettiers docs and on how I think our code looks like in other projects. I don't have strong opinions on any of these configurations. If you do - feel free to suggest changes!

## Checklist
- [x] Code builds and runs
- [ ] Tests have been added/updated
- [x] Documentation updated (if needed)

HenrikeW committed:
chore: add prettier as dependency

HenrikeW committed:
chore: add prettier config file

HenrikeW committed:
chore: add prettierignore file

HenrikeW committed:
docs: add documentation for prettier

KattisLej commented:
I think this looks good, the read me was clear and I could follow the instructions. There are some conflicts which I guess come from changes in develop without the prettier formatting.

pdmnyberg reviewed (changes requested):
I think this PR would be best served by not including the the application of `prettier`. That way it is easier to discuss the setup separate from the effects. (I have a faint memory that I might have said that it doesn't really matter but looking at this I think it makes sense to have the global application separately)

I would also like to suggest that we add a `scripts` entry in `package.json` that looks as follows `"prettify": "prettier --write ."`. 

Other than that I think this looks great!

HenrikeW committed:
chore: add prettify as npm script

HenrikeW commented:
> I think this PR would be best served by not including the the application of `prettier`. That way it is easier to discuss the setup separate from the effects. (I have a faint memory that I might have said that it doesn't really matter but looking at this I think it makes sense to have the global application separately)
> 
> I would also like to suggest that we add a `scripts` entry in `package.json` that looks as follows `"prettify": "prettier --write ."`.
> 
> Other than that I think this looks great!

Good point! Now I removed all commits that applied prettier on the code and added the line in `scripts` that you suggested.

pdmnyberg reviewed (approved):
LGTM!
